docs/architecture.md

```diff
@@ -4,213 +4,217 @@
+
+### Queues overview
+
+| name      | type     | batch | window | DLQ |
+|-----------|----------|-------|--------|-----|
+| piece     | standard | 10000 | 300 s  | TBD |
+| buffer    | FIFO     | 2     | 300 s  | TBD |
+| aggregate | FIFO     | 1     | 300 s  | TBD |
+
+Other relevant notes:
+- with the approach above, we have an append only log where database writes only need to happen when information leaves or gets into the system. This results in a quite small number of operations on the DB.
+- while `w3up` CAR files can be limited to `4 GiB` to have a better utilization of Fil sector space, same does not currently happen with `pickup` (and perhaps other systems in the future). Designing assuming an upper limit is not a good way to go in this system.
+- current design enables us to quite easily support bigger deals in the future.
+- if an aggregate fails to land into a Storage Provider, the problematic piece(s) can be removed and a buffer can be created without the problamtic piece(s). This way, it can already be added to the `buffer-queue`
+- we can support different producers (e.g. web3.storage, nft.storage, ...), as well as different requirements of aggregate building and aggregate submissions (e.g. different SLA requirements) by relying on SQS `message group id`
+- having a `deal-tracking` job instead of an additional queue gives us some additional benefits:
+  - we can have bigger intervals between each run compared to a maximum timeout of **300 seconds** for a queue consumer, enabling us to decrease number of requests
+  - we can keep track not only of deal resolution, but also instrument alerts when a status is `OFFERED` for more time than expected
+
+## Data Stack
+
+The Data stack is responsible for the state of the w3filecoin.
+
+It keeps track of the received pieces, submitted aggregates, as well as in which aggregate a given piece is (inclusion). In addition, it must keep the necessary state for the `buffer-queue` to operate, as well as the state of a given aggregate over time until a deal is fulfilled. It is worth mentioning, that keeping track of problematic pieces that could not be added to an aggregate should also be properly tracked.
+
+To achieve required state management, we will be relying on a S3 Bucket and a dynamoDB table as follows:
+
+| store           | type     | key                      | expiration |
+|-----------------|----------|--------------------------|------------|
+| piece-store     | DynamoDB | `piece`                  | ---        |
+| buffer-store    | S3       | `${blockCid}/{blockCid}` | 30 days    |
+| aggregate-store | DynamoDB | `piece`                  | ---        |
+| inclusion-store | DynamoDB | `aggregate`              | ---        |
+
+### Queries and associayed stores
+
+1. Check if piece is already in the pipeline (`inclusion-store` with fallback for `piece-store`)
+  - Before getting a piece into the pipelined queue, it should be stored to guarantee uniqueness
```

Comment:
We should consider the group / storefront here as well so that piece in one does not prevent it from been added into the other.